src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -1,500 +0,0 @@
- *
- * @param {String} newAmount
- * @returns {Boolean}
- */
-const validateAmount = (newAmount) => {
-    const decimalNumberRegex = new RegExp(/^\d+(,\d+)*(\.\d{0,2})?$/, 'i');
-    return newAmount === '' || (decimalNumberRegex.test(newAmount) && calculateAmountLength(newAmount) <= CONST.IOU.AMOUNT_MAX_LENGTH);
-};
-
-/**
- * Replaces each character by calling `convertFn`. If `convertFn` throws an error, then
- * the original character will be preserved.
- *
- * @param {String} text
- * @param {Function} convertFn - `fromLocaleDigit` or `toLocaleDigit`
- * @returns {String}
- */
-const replaceAllDigits = (text, convertFn) =>
-    _.chain([...text])
-        .map((char) => {
-            try {
-                return convertFn(char);
-            } catch {
-                return char;
-            }
-        })
-        .join('')
-        .value();
-
-function MoneyRequestAmountPage(props) {
-    const {translate, toLocaleDigit, fromLocaleDigit, numberFormat} = useLocalize();
-    const selectedAmountAsString = props.iou.amount ? CurrencyUtils.convertToWholeUnit(props.iou.currency, props.iou.amount).toString() : '';
-
-    const prevMoneyRequestID = useRef(props.iou.id);
-    const textInput = useRef(null);
-    const iouType = useRef(lodashGet(props.route, 'params.iouType', ''));
-    const reportID = useRef(lodashGet(props.route, 'params.reportID', ''));
-    const isEditing = useRef(lodashGet(props.route, 'path', '').includes('amount'));
-
-    const [amount, setAmount] = useState(selectedAmountAsString);
-    const [selectedCurrencyCode, setSelectedCurrencyCode] = useState(props.iou.currency);
-    const [shouldUpdateSelection, setShouldUpdateSelection] = useState(true);
-    const [selection, setSelection] = useState({start: selectedAmountAsString.length, end: selectedAmountAsString.length});
-
-    /**
-     * Event occurs when a user presses a mouse button over an DOM element.
-     *
-     * @param {Event} event
-     * @param {Array<string>} nativeIds
-     */
-    const onMouseDown = (event, nativeIds) => {
-        const relatedTargetId = lodashGet(event, 'nativeEvent.target.id');
-        if (!_.contains(nativeIds, relatedTargetId)) {
-            return;
-        }
-        event.preventDefault();
-        if (!textInput.current.isFocused()) {
-            textInput.current.focus();
-        }
-    };
-
-    const title = {
-        [CONST.IOU.MONEY_REQUEST_TYPE.REQUEST]: translate('iou.requestMoney'),
-        [CONST.IOU.MONEY_REQUEST_TYPE.SEND]: translate('iou.sendMoney'),
-        [CONST.IOU.MONEY_REQUEST_TYPE.SPLIT]: translate('iou.splitBill'),
-    };
-    const titleForStep = isEditing.current ? translate('iou.amount') : title[iouType.current];
-
-    /**
-     * Check and dismiss modal
-     */
-    useEffect(() => {
-        if (!ReportUtils.shouldHideComposer(props.report, props.errors)) {
-            return;
-        }
-        Navigation.dismissModal(reportID.current);
-    }, [props.errors, props.report]);
```

Comment:
Is logic no longer necessary? It was added in https://github.com/Expensify/App/pull/22198

Reply:
Ok, I will restore this logic as we will be able to access error prop https://github.com/Expensify/App/pull/23394

Reply:
Related PR https://github.com/Expensify/App/pull/23394. It's probably best to keep the logic